### PR TITLE
support fused_mutli_transformer_xpu_int8

### PR DESCRIPTION
**Pull request by NALLEIN**
### PR types
New features

### PR changes

OPs
### Description

Support  fused_multi_transformer_ptq op for inference, optimize the performance of beam_search decode by optimizing the gather operator.

### CODE REVIEW

File: paddle/phi/kernels/fusion/xpu/fused_multi_transformer_int8_xpu_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+              cache_kv_out[i]->numel());
+          PADDLE_ENFORCE_XDNN_SUCCESS(r, "xpu::copy");
+        }
+        // std::cout<<"curr——time_step: "<<time_step_value<<std::endl;
+        // std::cout<<"cache_kv_dims: "<<cache_kv_dims.to_str()<<std::endl;
+        // std::cout<<"cache_kv_data ptr: "<<cache_kv_data<<std::endl;
+        // std::cout<<"cache_kv_gather_dims:
+        // "<<cache_kv_gather_dims.to_str()<<std::endl; std::cout<<"cache_kv_out
+        // ptr: "<<reinterpret_cast<XPUTypeT*>(ctx.template
+        // Alloc<T>(cache_kv_out[i]))<<std::endl;
```
Comment (zhupengyang) [inline paddle/phi/kernels/fusion/xpu/fused_multi_transformer_int8_xpu_kernel.cc]: 建议删除debug信息

File: paddle/phi/kernels/fusion/xpu/fused_multi_transformer_int8_xpu_kernel.cc [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+              gather_index_t->dims().size() == 0 ? 1
+                                                 : gather_index_t->dims()[0],
+              gather_axis,
+              context_len_axis,
+              context_len,
+              time_step_value - context_len);
+          PADDLE_ENFORCE_XDNN_SUCCESS(r, "xpu::gather_inplace");
+        }
+        // std::cout<<"gather_part cache_kv_gather_dims:
+        // "<<cache_kv_gather_dims.to_str()<<std::endl;
```
Comment (zhupengyang) [inline paddle/phi/kernels/fusion/xpu/fused_multi_transformer_int8_xpu_kernel.cc]: 建议删除debug信息

File: paddle/phi/kernels/fusion/xpu/fused_multi_transformer_int8_xpu_kernel.cc (hunk 1 above)
Comment (NALLEIN) [reply paddle/phi/kernels/fusion/xpu/fused_multi_transformer_int8_xpu_kernel.cc]: Done.

**Review by zhiqiu — APPROVED** [review]
LGTM for const_cast

**Review by zhupengyang — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by From00 — APPROVED** [review]

**Pull request merged**